cnd/src/main.rs

```diff
@@ -160,14 +165,19 @@ fn version() {
     println!("{} {} ({})", name, version, short);
 }
 
+fn port_is_available(port: u16) -> bool {
+    match TcpListener::bind(("127.0.0.1", port)) {
```

Comment:
I don't think this is entirely correct. A machine can have several interfaces (and support several ip protocols). In the config file, you can specify a full socket. If you were to have to check if a port is available you'd need to use the socket instead of assume it tries to bind to the loopback ipv4 address.

Reply:
Good catch, you are indeed correct.